tests/console/generate_dud.pm

```diff
@@ -13,23 +13,75 @@ use strict;
 use warnings;
 use base "opensusebasetest";
 use testapi;
-use utils qw(zypper_call);
+use XML::Writer;
+use util qw(zypper_call);
 use autoyast qw(expand_variables);
 use registration qw(add_suseconnect_product get_addon_fullname);
 
+sub generate_xml {
+    my $addons = get_var('ADD_ON_PRODUCTS', '');
+
+    # Generate addon products xml file
+    my $writer = XML::Writer->new(
+        DATA_MODE => 'true',
+        DATA_INDENT => 2,
+        OUTPUT => "self"
+    );
+    $writer->startTag(
+        "add_on_products",
+        xmlns => "http://www.suse.com/1.0/yast2ns",
+        "xmlns:config" => "http://www.suse.com/1.0/configns"
+    );
+    $writer->startTag("product_items", "config:type" => "list");
+    for my $addon (split(/,/, $addons)) {
+        my @n = split(/\//, $addon);
+        my $name = $n[-1] . "_" . $n[-2];
```

Comment:
we had this regex working since some time ago and avoid problems with index out of range (index should be avoided as much as possible for code maintenance reasons):
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/data/autoyast_sle15/create_hdd/create_hdd_maintenance.xml.ep#L36